TensorFlow v2.8.0 fails test_simple_tensor_op on macOS

### Summary

[`tensorflow` `v2.8.0`](https://github.com/tensorflow/tensorflow/releases/tag/v2.8.0) is causing an `AssertionError` on macOS only because of floating point disagreement in 

https://github.com/scikit-hep/pyhf/blob/270d59d3900fdef7549015db45260e85f83e33b6/tests/test_tensor.py#L54

```pytb
=================================== FAILURES ===================================
______________________ test_simple_tensor_ops[tensorflow] ______________________

backend = (<pyhf.tensor.tensorflow_backend.tensorflow_backend object at 0x11ab80580>, None)

    def test_simple_tensor_ops(backend):
        tb = pyhf.tensorlib
        assert tb.tolist(tb.astensor([1, 2, 3]) + tb.astensor([4, 5, 6])) == [5, 7, 9]
        assert tb.tolist(tb.astensor([1]) + tb.astensor([4, 5, 6])) == [5, 6, 7]
        assert tb.tolist(tb.astensor([1, 2, 3]) - tb.astensor([4, 5, 6])) == [-3, -3, -3]
        assert tb.tolist(tb.astensor([4, 5, 6]) - tb.astensor([1])) == [3, 4, 5]
        assert tb.tolist(tb.sum(tb.astensor([[1, 2, 3], [4, 5, 6]]), axis=0)) == [5, 7, 9]
        assert tb.tolist(tb.product(tb.astensor([[1, 2, 3], [4, 5, 6]]), axis=0)) == [
            4,
            10,
            18,
        ]
        assert tb.tolist(tb.power(tb.astensor([1, 2, 3]), tb.astensor([1, 2, 3]))) == [
            1,
            4,
            27,
        ]
        assert tb.tolist(tb.divide(tb.astensor([4, 9, 16]), tb.astensor([2, 3, 4]))) == [
            2,
            3,
            4,
        ]
        assert tb.tolist(tb.sqrt(tb.astensor([4, 9, 16]))) == [2, 3, 4]
>       assert tb.tolist(tb.log(tb.exp(tb.astensor([2, 3, 4])))) == [2, 3, 4]
E       assert [1.9999999999999998, 3.0, 4.0] == [2, 3, 4]
E         At index 0 diff: 1.9999999999999998 != 2
E         Use -v to get the full diff

tests/test_tensor.py:54: AssertionError
```

This wasn't caught in the ["HEAD of dependencies" workflow](https://github.com/scikit-hep/pyhf/blob/270d59d3900fdef7549015db45260e85f83e33b6/.github/workflows/dependencies-head.yml) in advance as we're currently only testing on ubuntu there.

### OS / Environment

Seen in GitHub Actions CI with

```
Operating System
  macOS
  11.6.2
  20G314
Virtual Environment
  Environment: macos-11
  Version: 20220124.1
  Included Software: https://github.com/actions/virtual-environments/blob/macOS-11/20220124.1/images/macos/macos-11-Readme.md
  Image Release: https://github.com/actions/virtual-environments/releases/tag/macOS-11%2F20220124.1
```

[permalink to included software](https://github.com/actions/virtual-environments/blob/853e7cb28ef5acd0fa43cd03c042d8b2ace675f6/images/macos/macos-11-Readme.md)

### Steps to Reproduce

On macOS with the current `HEAD` dev version of `pyhf` installed with dev extra dependencies

<!--- ...or if you have a failing CLI command paste it between the quotes below -->
```console (paste below)
$ python -m pip install --upgrade 'tensorflow==2.8.0'
$ pytest -sx tests/test_tensor.py -k test_simple_tensor_ops[tensorflow]
```


### File Upload (optional)

_No response_

### Expected Results

The tests to pass without error.

### Actual Results

Fails with assertion error

```pytb
>       assert tb.tolist(tb.log(tb.exp(tb.astensor([2, 3, 4])))) == [2, 3, 4]
E       assert [1.9999999999999998, 3.0, 4.0] == [2, 3, 4]
E         At index 0 diff: 1.9999999999999998 != 2
E         Use -v to get the full diff

tests/test_tensor.py:54: AssertionError
```


### pyhf Version

`HEAD` is currently at 270d59d3900fdef7549015db45260e85f83e33b6


### Code of Conduct

- [X] I agree to follow the Code of Conduct

--- a/tests/test_tensor.py
+++ b/tests/test_tensor.py
@@ -1,3 +1,4 @@
+from sys import platform
 import pytest
 import logging
 import numpy as np
@@ -87,6 +88,17 @@ def test_simple_tensor_ops(backend):
         [2.0, 5.0],
         [3.0, 6.0],
     ]
+
+
+@pytest.mark.xfail(platform == "darwin", reason="c.f. Issue #1759")
+@pytest.mark.only_tensorflow
+def test_simple_tensor_ops_floating_point(backend):
+    """
+    xfail test to know if test_simple_tensor_ops stops failing for tensorflow
+    on macos
+    """
+    tb = pyhf.tensorlib
+    assert tb.tolist(tb.log(tb.exp(tb.astensor([2, 3, 4])))) == [2, 3, 4]
 
 
 def test_tensor_where_scalar(backend):

--- a/tests/test_tensor.py
+++ b/tests/test_tensor.py
@@ -51,7 +51,10 @@ def test_simple_tensor_ops(backend):
         4,
     ]
     assert tb.tolist(tb.sqrt(tb.astensor([4, 9, 16]))) == [2, 3, 4]
-    assert tb.tolist(tb.log(tb.exp(tb.astensor([2, 3, 4])))) == [2, 3, 4]
+    # c.f. Issue #1759
+    assert tb.tolist(tb.log(tb.exp(tb.astensor([2, 3, 4])))) == pytest.approx(
+        [2, 3, 4], 1e-9
+    )
     assert tb.tolist(tb.abs(tb.astensor([-1, -2]))) == [1, 2]
     assert tb.tolist(tb.erf(tb.astensor([-2.0, -1.0, 0.0, 1.0, 2.0]))) == pytest.approx(
         [

